addons/mass_mailing/models/mailing.py

```diff
@@ -1335,6 +1342,52 @@ def _image_to_url(b64image: bytes):
                 node.attrib['src'] = _image_to_url(image)
                 modified = True
 
+    def _crop_vml_images(self, body_html):
+        """
+        Find VML v:image elements, crop their source images, make an attachement
+        out of them and replace their source with an url to the attachement.
+        """
+
+        def _image_to_url(b64image: bytes):
+            """Store an image in an attachement and returns an url"""
+            attachment = self.env['ir.attachment'].create({
+                'datas': b64image,
+                'name': "cropped_image_mailing_{}".format(self.id),
+                'type': 'binary',})
+
+            attachment.generate_access_token()
+
+            return '/web/image/%s?access_token=%s' % (
+                attachment.id, attachment.access_token)
```

Comment:
Is there a reason why this function is inside the other one ? It doesn't look to be scoped by any parameter of the other one.

Reply:
Done.